app/models/institutions_archive.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require_dependency 'institution'
+
+class InstitutionsArchive < Institution
+  self.table_name = 'institutions_archives'
+
+  # class methods
+  def self.archive(version)
+    version_number = version.number
+
+    conn = ActiveRecord::Base.connection
+
+    begin
+      ActiveRecord::Base.transaction do
+        create_archives(version_number, conn)
+        cleanup_institutions(version_number, conn)
+      end
+    rescue ActiveRecord::StatementInvalid => e
+      notice = 'There was an error occurring at the database level'
+      error_msg = e.original_exception.result.error_message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    rescue StandardError => e
+      notice = 'There was an error of unexpected origin'
+      error_msg = e.message
+      Rails.logger.error "#{notice}: #{error_msg}"
+    end
```

Comment:
If this is something we will run often, we may consider adding further logging (i.e. to Sentry)

Reply:
Just looking back over the past 7 months, it looks like new production versions are build 1 - 4 times/mo (which will be the impetus for this to run)

Reply:
Added sentry logging